### PR TITLE
feat(server): reduce progress update interval during the game

### LINKED ISSUE #3: [FEATURE] reduce the progress update interval during game
During the game, the progress is updated on every keystroke. This should be avoided, instead the progress should be updated whenever the progress increases by 5%. For the last 10% of progress update, the update to server should happen on every key stroke, this is in order have greater accuracy in determining the winner.

### PR DESCRIPTION
**Pull request by Abhicodes-crypto**
Address #3

### CODE REVIEW

**Comment by Abhicodes-crypto** [comment]
Hey @Narayanbhat166 , Can you review this ?

File: src/server_utils/message_handlers.rs [hunk 1]
```diff
@@ -75,10 +75,17 @@ pub async fn handle_client_messages(
...
         models::WSClientMessage::UpdateProgress { game_id, progress } => {
-            db.update_game_progress(&game_id, current_user_id, progress)
-                .await;
+            let current_progress = db.find_game_progress(&game_id, current_user_id).await;
+            
+            if progress < 70 && progress - current_progress < 5 {
+                eprintln!("Skipping update progress of game_id: {game_id}, current_user_id: {current_user_id}, progress: {progress}");
+            } else {
+                db.update_game_progress(&game_id, current_user_id, progress)
+                    .await;
```
Comment (Narayanbhat166) [inline src/server_utils/message_handlers.rs]: can we update the game progress every time, but broadcast it to all the other users only if the progress is significant ( the if condition ). Some updates might not be persisted in the case of partial updates in the proposed implementation

Comment (Abhicodes-crypto) [reply src/server_utils/message_handlers.rs]: This was done.

Comment (Narayanbhat166) [reply src/server_utils/message_handlers.rs]: This project has been moved to https://github.com/Narayanbhat166/blazer, however it is still under development. You can feel free to contribute. Let me know if you need any help.

**Pull request merged**